src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorResolver.cs

```diff
@@ -104,26 +112,85 @@ public IEnumerable<TagHelperDescriptor> Resolve([NotNull] TagHelperDescriptorRes
         /// <param name="assemblyName">
         /// The name of the assembly to resolve <see cref="TagHelperDescriptor"/>s from.
         /// </param>
+        /// <param name="prefix">The prefix to the created <see cref="TagHelperDescriptor"/>s.</param>
         /// <param name="documentLocation">The <see cref="SourceLocation"/> of the directive.</param>
         /// <param name="errorSink">Used to record errors found when resolving <see cref="TagHelperDescriptor"/>s 
         /// within the given <paramref name="assemblyName"/>.</param>
         /// <returns><see cref="TagHelperDescriptor"/>s for <see cref="ITagHelper"/>s from the given
         /// <paramref name="assemblyName"/>.</returns>
         // This is meant to be overridden by tooling to enable assembly level caching.
-        protected virtual IEnumerable<TagHelperDescriptor> ResolveDescriptorsInAssembly(string assemblyName,
-                                                                                        SourceLocation documentLocation,
-                                                                                        ParserErrorSink errorSink)
+        protected virtual IEnumerable<TagHelperDescriptor> ResolveDescriptorsInAssembly(
+            string assemblyName,
+            string prefix,
+            SourceLocation documentLocation,
+            ParserErrorSink errorSink)
         {
             // Resolve valid tag helper types from the assembly.
             var tagHelperTypes = _typeResolver.Resolve(assemblyName, documentLocation, errorSink);
 
             // Convert types to TagHelperDescriptors
             var descriptors = tagHelperTypes.SelectMany(
-                type => TagHelperDescriptorFactory.CreateDescriptors(assemblyName, type));
+                type => TagHelperDescriptorFactory.CreateDescriptors(assemblyName, prefix, type));
 
             return descriptors;
         }
 
+        private static string ResolveTagHelperPrefix(TagHelperDescriptorResolutionContext context)
+        {
+            var prefixDirectiveDescriptors = context.DirectiveDescriptors.Where(
+                descriptor => descriptor.DirectiveType == TagHelperDirectiveType.TagHelperPrefix);
+
+            if (prefixDirectiveDescriptors.Count() > 1)
```

Comment:
this extra condition is unnecessary since `Skip(1)` will simply return an empty enumeration if `prefixDirectiveDescriptors` is empty or contains one item.